### ISSUE TITLE
Problem with updating user-changeable properties

### ISSUE THREAD

**Issue by cornerman** [OutWatch/outwatch#105]
Before scala-dom-types, we used the `value` attribute and manually synced this with property `value` (valueSyncHook). This was specific for the `value` property and still left problems with e.g. `checked`. Now, we use the property `value` directly and not the attribute. This generally fixes the problem, but now we are running in into an issue with updating user-changeable properties in snabbdom: https://github.com/snabbdom/snabbdom/issues/345.

**Comment by mariusmuja** [comment]
React.js has the notion of `controlled`/`un-controlled` components, the difference being the source of truth about the state of the component (the DOM or the component).

For example translating the example from snabbdom/snabbdom#345 to Outwatch it could be something like:

```scala
import outwatch.dom.{Handler, Attributes => A, Tags => T}

def component(): VNode = {
  Handler.create[String].flatMap { strHandler =>
    T.div(
      T.input(A.`type` := "text", A.value := "test", A.value <-- strHandler),
      T.button(A.onClick("test") --> strHandler, "Set")
    )
  }
}
```

Here, pressing the "Set" button after changing the input text doesn't do anything because the the new prop'a value is the same as the old prop.

We can turn this into a "controlled component" (in react.js terminology), by sync-ing the DOM state to the component, which fixes the issue:

```scala
def component(): VNode = {
  Handler.create[String].flatMap { strHandler =>
    T.div(
      T.input(A.`type` := "text", A.value := "test", A.value <-- strHandler, A.onInputString --> strHandler),
      T.button(A.onClick("test") --> strHandler, "Set")
    )
  }
}
```

For a nicer syntax, we could have a `valueControlled` VDomModifier builder:

```scala
object valueControlled {
  def <-->(handler: Handler[String]): VDomModifier = Seq(
    A.value <-- handler, A.onInputString --> handler
  )
}

def component(): VNode = {
  Handler.create[String].flatMap { strHandler =>
    T.div(
      T.input(A.`type` := "text", A.value := "test", valueControlled <--> strHandler),
      T.button(A.onClick("test") --> strHandler, "Set")
    )
  }
}

```

**Comment by fdietze** [comment]
I'm also in favor of not expecting too much knowledge about virtual dom inner workings from the user. It should be simple and just work.

I researched a bit and never found any benchmark or evidence that reading dom properties may be slow. Does anyone know something more about that? Benchmarks would be nice.

When looking at constructs like

```scala
A.value <-- strHandler, A.onInputString --> strHandler
```

I doubt that updating the dom on every keystroke is better than checking once if if the dom property changed.

**Comment by cornerman** [comment]
fixed by 441298a2ee57f539cc00cf5eba7ddd3e1cfc91c6
